### ISSUE TITLE
Add a Benchmarking / profiling example

### ISSUE THREAD

**Issue by lebrice** [mila-iqia/ResearchTemplate#11]
Should be completed after  https://github.com/mila-iqia/mila-docs/issues/247

Now that we have an example of how to benchmark the throughput and identify bottlenecks in the mila-docs, the research project template should also make this easy to do.

- [ ] Add an example experiment configuration and accompanying notebook, that use the pytorch profiler and does the same kind of profiling as in the example, but using the template 
- [ ] Add an example of a sweep over some parameters, with the training throughput as the metric, and using different kinds of GPUs.
- [ ] Create a wandb report with the throughput comparison between the different GPU types.
  - 1. Find the best datamodule parameters to maximize the throughput (batches per second) without training (NoOP algo)
  - 2. Measure the performance on different GPUS using the optimal datamodule params from before (and keeping other parameters the same)
  - 3. Using the results from before, do a simple sweep over model hyper-parameters to maximize the utilization of the selected GPU  (which was selected as a tradeoff between performance and difficulty to get an allocation). For example if the RTX8000's are 20% slower than A100s but 5x easier to get an allocation on, use those instead.
- [ ] If done after DRAC support, also include a comparison between Mila/DRAC clusters. (For example, the optimal num_workers might be greater in DRAC due to the very slow $SCRATCH filesystems, could be interesting to take a look at that).

**Comment by lebrice** [comment]
Also interesting: https://github.com/nschloe/tuna

**Comment by lebrice** [comment]
An example of step 7+ would be something like this: https://pytorch.org/blog/accelerating-generative-ai/
